runtime/README.md

```diff
@@ -692,107 +708,504 @@ Command Code: `0x434B_4558` ("CKEX")
-| auth_req_result | u32      | AUTHORIZE_IMAGE: 0xDEADC0DE and DENY_IMAGE_AUTHORIZATION: 0x21523F21    |
+| **Name**        | **Type** | **Description**                                                            |
+| --------------- | -------- | -------------------------------------------------------------------------- |
+| chksum          | u32      | Checksum over other output arguments, computed by Caliptra. Little endian. |
+| fips_status     | u32      | Indicates if the command is FIPS approved or an error.                     |
+| auth_req_result | u32      | Authorized: `0xDEADC0DE` |
+|                 |          | Denied: `0x21523F21`     |
+
+### SET_IMAGE_METADATA
+
+Command Code: `0x5349_4D44` ("SIMD")
+
+*Table: `SET_IMAGE_METADATA` input arguments*
+
+| **Name**    | **Type** | **Description**                                                                     |
+| ----------- | -------- | ----------------------------------------------------------------------------------- |
+| chksum      | u32      | Checksum over other input arguments, computed by the caller. Little endian.         |
+| metadata size    | u32    |                                     |
+| metadata | u8[...]   | Metadata                                                                  |
+
+*Table: `SET_IMAGE_METADATA` output arguments*
+| **Name**        | **Type** | **Description**                                                            |
+| --------------- | -------- | -------------------------------------------------------------------------- |
+| chksum          | u32      | Checksum over other output arguments, computed by Caliptra. Little endian. |
+| fips_status     | u32      | Indicates if the command is FIPS approved or an error.                     |
+
+
+## Mailbox commands: Cryptographic Mailbox (2.0)
+
+These commands are used by the [Cryptograhic Mailbox](#cryptographic-mailbox-commands-20) system.
+
+### CM_SHA
+
+This computes SHA hashes of data, which may be larger than a single mailbox command allows. It also supports additional algorithms.
+
+Command Code: `0x434D_5348` ("CMSH")
+
+*Table: `CM_SHA` input arguments*
+
+| **Name**       | **Type** | **Description**                   |
+| -------------- | -------- | --------------------------------- |
+| chksum         | u32      |                                   |
+| flags          | u32      | Bit 0 = this is the final message |
+| hash algorithm | u32      | Enum.                             |
+|                |          | Value 0 = reserved                |
+|                |          | Value 1 = SHA2-256                |
+|                |          | Value 2 = SHA2-384                |
+|                |          | Value 3 = SHA2-512                |
+| context size   | u32      | Size of the context               |
+| context        | u8[...]  |                                   |
```

Comment:
Can the context be fixed-size? Since we use zerocopy for command serialization/deserialization, it's quite a bit simpler to handle if the interior of the request is fixed-size.

Reply:
For this particular one, yes, but I'd prefer to be flexible so that we don't have to change the format of the command if we decide to support other algorithms.